pkg/kubectl/genericclioptions/print_flags.go

```diff
@@ -55,27 +57,42 @@ func IsNoCompatiblePrinterError(err error) bool {
 // used across all commands, and provides a method
 // of retrieving a known printer based on flag values provided.
 type PrintFlags struct {
-	JSONYamlPrintFlags *JSONYamlPrintFlags
-	NamePrintFlags     *NamePrintFlags
+	JSONYamlPrintFlags   *JSONYamlPrintFlags
+	NamePrintFlags       *NamePrintFlags
+	TemplatePrinterFlags *KubeTemplatePrintFlags
 
 	TypeSetterPrinter *printers.TypeSetterPrinter
 
-	OutputFormat *string
+	OutputFormat    *string
+	outputFlag      *pflag.Flag
+	outputDefaulted bool
 }
 
 func (f *PrintFlags) Complete(successTemplate string) error {
 	return f.NamePrintFlags.Complete(successTemplate)
 }
 
 func (f *PrintFlags) AllowedFormats() []string {
-	return append(f.JSONYamlPrintFlags.AllowedFormats(), f.NamePrintFlags.AllowedFormats()...)
+	ret := []string{}
+	ret = append(ret, f.JSONYamlPrintFlags.AllowedFormats()...)
+	ret = append(ret, f.NamePrintFlags.AllowedFormats()...)
+	ret = append(ret, f.TemplatePrinterFlags.AllowedFormats()...)
+	return ret
 }
 
 func (f *PrintFlags) ToPrinter() (printers.ResourcePrinter, error) {
 	outputFormat := ""
 	if f.OutputFormat != nil {
 		outputFormat = *f.OutputFormat
 	}
+	// For backwards compatibility we want to support a --template argument given, even when no --output format is provided.
+	// If a default output format has been set, but no explicit output format has been provided via the --output flag, fallback
+	// to honoring the --template argument.
+	if f.TemplatePrinterFlags != nil && f.TemplatePrinterFlags.TemplateArgument != nil &&
+		len(*f.TemplatePrinterFlags.TemplateArgument) > 0 &&
+		(len(outputFormat) == 0 || (f.outputDefaulted && !f.outputFlag.Changed)) {
```

Comment:
doesn't this change make it impossible to compose this without specifically adding these flags?